src/core/ddsi/src/q_entity.c

```diff
@@ -1463,14 +1488,14 @@ static void writer_drop_local_connection (const struct ddsi_guid *wr_guid, struc
   }
 }
 
-static void reader_drop_connection (const struct ddsi_guid *rd_guid, const struct proxy_writer *pwr)
+static void reader_drop_connection (const struct ddsi_guid *rd_guid, const struct proxy_writer *pwr, bool unmatch)
```

Comment:
Replying to myself after considering the two points I raised: that race between transitions from alive to not-alive and vice-versa cannot exist because the proxy writer lock is held for the duration of the transition.

Holding the proxy writer's lock while calling listeners is not ideal, but that same problem already exists for other listeners. The argument for invoking listeners inline is that it gives the lowest possible latency as well as a number of guarantees you cannot otherwise have, and the price you pay is that you have to be careful what you do in a listener. This is not something to make a fuss over that here & now.

The other point, delete/loss-of-liveliness in parallel I haven't yet tried to figure out in detail. But I have decided that will not block acceptance of the pull request.